core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -750,7 +750,7 @@ private void fillTriggeredElements(Triggerable t,
                 // when we grabbed all triggerables that target children of the
                 // repeat entry (via initTriggerablesRootedBy). Hence skip them
                 if (!isRepeatEntryInit && t.isCascadingToChildren()) {
-                    updatedNodes = findCascadeReferences(target, updatedNodes);
+                    //updatedNodes = findCascadeReferences(target, updatedNodes);
```

Comment:
@phillipm Had me try commenting out a few different lines in his optimizations to see if that fixed it, but none of these changes had an impact.